hotspot/src/share/vm/opto/addnode.hpp

```diff
@@ -279,4 +279,38 @@ class MinINode : public MaxNode {
   virtual Node *Ideal(PhaseGVN *phase, bool can_reshape);
 };
 
+//------------------------------MaxLNode---------------------------------------
+// MAXimum of 2 longs.
+class MaxLNode : public MaxNode {
+public:
+  MaxLNode(Compile* C, Node* in1, Node* in2) : MaxNode(in1, in2) {
+    init_flags(Flag_is_macro);
+    C->add_macro_node(this);
+  }
+  virtual int Opcode() const;
+  virtual const Type* add_ring(const Type* t0, const Type* t1) const;
+  virtual const Type* add_id() const { return TypeLong::make(min_jlong); }
+  virtual const Type* bottom_type() const { return TypeLong::LONG; }
+  virtual uint ideal_reg() const { return Op_RegL; }
+  virtual Node* Identity(PhaseGVN* phase);
```

Comment:
Is it OK that `MaxLNode::Identity` has different signature than `AddNode::Identity`?
```
  virtual Node *Identity( PhaseTransform *phase );
```
Tha same question is for MinLNode and ConvI2LNode.

Reply:
Yes, as [JDK-8146629](https://bugs.openjdk.org/browse/JDK-8146629 "Make phase->is_IterGVN() accessible from Node::Identity and Node::Value") (openjdk/jdk11u@69b52aa28be9054ea53bf9587ef7b563a658e510) doesn't have a 8u backport, it would have been neater to have `PhaseTransform* phase`.

What happened here is that neither the patch application nor the compilation failed, since `PhaseGVN` inherits from `PhaseValues`, which inherits from `PhaseTransform`.

However, there shouldn't be any issue, as the C++ type system is ensuring that all the callers are passing a `PhaseGVN` pointer.

Reply:
Compilation didn't fail because not all the warnings are enabled.

Enabling `-Woverloaded-virtual` causes a number of warnings like the following:
```
hotspot/src/share/vm/opto/addnode.hpp:50:17: error: ‘virtual Node* AddNode::Identity(PhaseTransform*)’ was hidden [-Werror=overloaded-virtual]
   50 |   virtual Node *Identity( PhaseTransform *phase );
hotspot/src/share/vm/opto/addnode.hpp:312:17: error:   by ‘virtual Node* MinLNode::Identity(PhaseGVN*)’ [-Werror=overloaded-virtual]
  312 |   virtual Node* Identity(PhaseGVN* phase);
```
that may point to a serious bug.

In few words, `MaxLNode::Identity` may be never called in case:
```
AddNode *obj = new MaxLNode;
obj->Identity(...); // AddNode::Identity() is called.
```

Reply:
You are right, I was missing the `MaxLNode`&larr;`MaxNode`&larr;`AddNode` inheritance and the fact that `MaxLNode::Identity(PhaseGVN* phase)` doesn't override but hides `AddNode::Identity(PhaseTransform* phase)`, sorry for the confusion.

After my comment, we continued analyzing it with @martinuy and arrived to similar conclusions. We'll be proposing a fix for this later today.

Reply:
As this pull request is already integrated, I have created the [JDK-8335552: \[8u\] JDK-8303466 backport to 8u requires 3 ::Identity signature fixes](https://bugs.openjdk.org/browse/JDK-8335552) follow up.